spark/v3.1/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java

```diff
@@ -301,17 +284,33 @@ public void testReplacePartition() {
     sql("ALTER TABLE %s ADD PARTITION FIELD days(ts)", tableName);
     table.refresh();
     PartitionSpec expected =
-        PartitionSpec.builderFor(table.schema()).withSpecId(1).day("ts").build();
```

Comment:
We do we need to do the change here? I'm not a big fan of have String constants for expected values, even if we are parsing them into a friendly object for comparison.

Reply:
I mentioned the reason why some tests refer to String constants [here](https://github.com/apache/iceberg/pull/8381#discussion_r1303610435).